### DISCUSSION TITLE
Connect to broker using server certificate, client certificate and client key

### DISCUSSION THREAD

**Discussion by timurig** [dotnet/MQTTnet#1821] [Q&A]
Hi,
I am having problem connecting to a broker using 3 files - server certificate, client certificate and client key. Can't find anything compatible in MQTTnet. Could somebody help me with that please?

**Comment by SeppPenner** [comment]
Did you look at the samples by any chance? https://github.com/dotnet/MQTTnet/blob/master/Samples/Client/Client_Connection_Samples.cs Maybe, this helps...

**Reply by timurig** [reply]
Actually I did, it was the reason of me asking the question in the first place. It turned out the encryption algorithm is the source of the problem. The same exact application with the same exact certificates doesn't work if run on Server 2012 machine, but works fine on Server 2019 or Windows 10. It looks like  encryption algorithm used by windows was changed (at about 2016) and there is no way to specify explicitly which one is to be used in your application (at least I couldn't find it). In this situation upgrading the OS is the only option.
